File: app/src/main/java/co/smartreceipts/android/model/impl/columns/receipts/ReceiptExchangeRateColumn.java
package co.smartreceipts.android.model.impl.columns.receipts;

import android.support.annotation.NonNull;

import co.smartreceipts.android.model.Receipt;
import co.smartreceipts.android.model.impl.columns.AbstractColumnImpl;
import co.smartreceipts.android.sync.model.SyncState;

/**
 * Provides a column that returns the category code for a particular receipt
 */
public final class ReceiptExchangeRateColumn extends AbstractColumnImpl<Receipt> {

    public ReceiptExchangeRateColumn(int id, @NonNull String name, @NonNull SyncState syncState, int customOrderId) {
        super(id, name, syncState, customOrderId);
    }

    @Override
    public String getValue(@NonNull Receipt receipt) {
        return receipt.getPrice().getExchangeRate().getDecimalFormattedExchangeRate(receipt.getTrip().getTripCurrency());
    }
}

File: app/src/main/java/co/smartreceipts/android/model/impl/columns/receipts/ReceiptExchangeRateColumn.kt
package co.smartreceipts.android.model.impl.columns.receipts

import co.smartreceipts.android.model.Receipt
import co.smartreceipts.android.model.impl.columns.AbstractColumnImpl
import co.smartreceipts.core.sync.model.SyncState
import java.util.*

/**
 * Provides a column that returns the category code for a particular receipt
 */
class ReceiptExchangeRateColumn(id: Int, syncState: SyncState, customOrderId: Long, uuid: UUID) :
    AbstractColumnImpl<Receipt>(
        id,
        ReceiptColumnDefinitions.ActualDefinition.EXCHANGE_RATE,
        syncState,
        customOrderId,
        uuid
    ) {

    override fun getValue(rowItem: Receipt): String =
        rowItem.price.exchangeRate.getDecimalFormattedExchangeRate(rowItem.trip.tripCurrency)
}
